rust/datafusion/src/execution/context.rs

```diff
@@ -287,6 +287,21 @@ impl ExecutionContext {
             .insert(name.to_string(), provider.into());
     }
 
+    /// Deregisters the named table.
+    /// 
+    /// Returns true if the table was successfully de-reregistered.
+    pub fn deregister_table(
+        &mut self,
+        name: &str
+    ) -> bool {
```

Comment:
I would consider raising an error if it cannot be un-registered instead of a boolean.

Reply:
I don't think it should be an error to attempt to deregister with an unknown table.

Reply:
my thinking was that as a user, if a function returns a `bool`, I am incentivized to write `let _ = ctx.deregister_table(table)` to ignore the value. If the function returns an error, I am incentivized to think about what they want to be doing if the table does not exist (typically a `unwrap` or `?`). So, imo, either a `Result<()>` or `()` seemed most reasonable.

Taking a step back, I can see that `register_table` currently just silently overwrites an existing table. So, for consistency, returning `()` makes sense. OTOH, I am not convinced that `register_table`s API is though through: a user can silently overwrite a table 🤯

Anyways, let's try to clean these later. I think that this PR is useful as is 👍

Reply:
There is precedent in Rust's stdlib for returning a boolean here (e.g. [`HashSet::remove`](https://doc.rust-lang.org/stable/std/collections/struct.HashSet.html#method.remove)), or an `Option<>` if it makes sense to reuse the removed `TableProvider`.

(also, unless the function is `#[must_use]`, you don't have to `let _ = ..` to ignore a boolean return)

I guess the question is whether accidentally failing to deregister a table can trigger an issue later on. Perhaps not? The user would have to execute a query against a table they thought had been deregistered. 

I agree that `register_table` should probably return a `Some(TableProvider)` when it replaces an existing table, though.